xbmc/addons/AddonInfo.h

```diff
@@ -85,13 +85,240 @@ namespace ADDON
   typedef std::map<std::string, std::string> InfoMap;
   typedef std::map<std::string, std::string> ArtMap;
 
+  class CAddonBuilder;
+
   class CAddonInfo
   {
   public:
     CAddonInfo();
     CAddonInfo(std::string id, TYPE type);
 
     /*!
+     * @brief To ask generated class is usable and all needed parts are set.
+     *
+     * @return True is OK and usable, otherwise false.
+     *
+     * @note This function is only be related to constructor's who read the
+     * addon xml part.
+     */
+    bool IsUsable() const { return m_usable; }
+
+    /*!
+     * @brief Parts used from ADDON::CAddonDatabase
+     */
+    //@{
+    void SetInstallDate(CDateTime installDate) { m_installDate = installDate; }
+    void SetLastUpdated(CDateTime lastUpdated) { m_lastUpdated = lastUpdated; }
+    void SetLastUsed(CDateTime lastUsed) { m_lastUsed = lastUsed; }
+    void SetOrigin(std::string origin) { m_origin = std::move(origin); }
+    //@}
+
+    /*!
+     * @brief Get the ID from add-on
+     *
+     * The ID is defined in addon.xml but should also equal with add-on path!
+     *
+     * @return ID string of add-on
+     */
+    const std::string& ID() const { return m_id; }
+
+    /*!
+     * @brief The master type id of add-on
+     *
+     * @note to get a human readable name for the type can be TranslateType(...)
+     * from here used.
+     *
+     * @return Master Type ID of add-on
+     */
+    TYPE MainType() const { return m_mainType; }
+
+    /*!
+     * @brief To get version of add-on
+     *
+     * @return The AddonVersion class from add-on
+     */
+    const AddonVersion& Version() const { return m_version; }
+
+    /*!
+     * @brief To get min version of add-on
+     *
+     * Used to check compatibility with others who request them
+     *
+     * @return The AddonVersion class from add-on with minimum version
+     */
+    const AddonVersion& MinVersion() const { return m_minversion; }
+
+    /*!
+     * @brief Human readable name of add-on
+     *
+     * @return Add-on name
+     */
+    const std::string& Name() const { return m_name; }
+
+    /*!
+     * @brief Licence text of add-on
+     *
+     * @return the complete licence text defined on addon.xml
+     */
```

Comment:
I find all these getter 'docstrings' very superficial.. It's completely obvious to everyone what all these methods does.